cmd/copy.go

```diff
@@ -1480,16 +1496,28 @@ func init() {
 		Example:    copyCmdExample,
 		Args: func(cmd *cobra.Command, args []string) error {
 			if len(args) == 1 { // redirection
-				if stdinPipeIn, err := isStdinPipeIn(); stdinPipeIn == true {
+				// Enforce the usage of from-to flag when pipes are involved
+				if raw.fromTo == "" {
+					return fmt.Errorf("fatal: from-to argument required, PipeBlob (upload) or BlobPipe (download) is acceptable")
+				}
+				var userFromTo common.FromTo
+				err := userFromTo.Parse(raw.fromTo)
+				if err != nil || (userFromTo != common.EFromTo.PipeBlob() && userFromTo != common.EFromTo.BlobPipe()) {
+					return fmt.Errorf("fatal: invalid from-to argument passed: %s", raw.fromTo)
+				}
+
+				if userFromTo == common.EFromTo.PipeBlob() {
+					// Case 1: PipeBlob. Check for the std input pipe
+					stdinPipeIn, err := isStdinPipeIn()
+					if stdinPipeIn == false || err != nil {
+						return fmt.Errorf("fatal: failed to read from Stdin due to error: %s", err)
```

Comment:
@mohsha-msft just to confirm, the error thrown here is on STDERR, correct?

Reply:
yes correct.